packages/nimble-components/src/wafer-map/modules/rendering.ts

```diff
@@ -1,34 +1,35 @@
+import type { WaferMap } from '..';
 import type { DieRenderInfo, Dimensions } from '../types';
-import type { DataManager } from './data-manager';
 
 /**
- * Responsible for drawing the dies inside the wafer map
+ * Responsible for drawing the dies inside the wafer map and scaling the canvas
  */
-export class RenderingModule {
+export class RenderingModule extends EventTarget {
     private readonly context: CanvasRenderingContext2D;
     private dieSize?: number;
     private readonly dies: DieRenderInfo[];
     private readonly dimensions: Dimensions;
     private readonly labelFontSize: number;
 
-    public constructor(waferData: DataManager, canvas: HTMLCanvasElement) {
-        this.context = canvas.getContext('2d')!;
-        this.dies = waferData.diesRenderInfo;
-        this.dimensions = waferData.dieDimensions;
-        this.labelFontSize = waferData.labelsFontSize;
+    public constructor(private readonly wafermap: WaferMap) {
+        super();
+        this.context = wafermap.canvas.getContext('2d')!;
+        this.dies = wafermap.dataManager!.diesRenderInfo;
+        this.dimensions = wafermap.dataManager!.dieDimensions;
+        this.labelFontSize = wafermap.dataManager!.labelsFontSize;
     }
 
-    public drawWafer(transform?: number): void {
-        this.renderDies(transform);
-        this.renderText(transform);
+    public drawWafer(): void {
+        this.context.save();
+        this.clearCanvas();
+        this.scaleCanvas();
+        this.renderDies();
+        this.renderText();
+        this.context.restore();
+        this.dispatchEvent(new CustomEvent('render-complete'));

```

Comment:
What is the goal of this event? It will potentially fire many times. i,e, continuously while zooming. We should not add it until it's needed so we can see the context.